File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/views/social/OldQuestProgressView.java
package com.habitrpg.android.habitica.ui.views.social;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.facebook.drawee.view.SimpleDraweeView;
import com.habitrpg.android.habitica.R;
import com.habitrpg.android.habitica.models.inventory.QuestCollect;
import com.habitrpg.android.habitica.models.inventory.QuestContent;
import com.habitrpg.android.habitica.models.inventory.QuestProgress;
import com.habitrpg.android.habitica.models.inventory.QuestProgressCollect;
import com.habitrpg.android.habitica.ui.helpers.DataBindingUtils;
import com.habitrpg.android.habitica.ui.views.ValueBar;

import butterknife.BindView;
import butterknife.ButterKnife;

public class OldQuestProgressView extends LinearLayout {

    @BindView(R.id.boss_name_view)
    TextView bossNameView;
    @BindView(R.id.boss_health_view)
    ValueBar bossHealthView;
    @BindView(R.id.collection_container)
    ViewGroup collectionContainer;

    public OldQuestProgressView(@NonNull Context context) {
        super(context);
        setupView(context);
    }

    public OldQuestProgressView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        setupView(context);
    }

    private void setupView(Context context) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        inflater.inflate(R.layout.quest_progress_old, this);

        ButterKnife.bind(this, this);
    }

    public void setData(QuestContent quest, QuestProgress progress) {
        collectionContainer.removeAllViews();
        if (quest.isBossQuest()) {
            bossNameView.setText(quest.getBoss().getName());
            if (progress != null) {
                bossHealthView.set(progress.getHp(), quest.getBoss().getHp());
            }
            bossNameView.setVisibility(View.VISIBLE);
            bossHealthView.setVisibility(View.VISIBLE);
        } else {
            bossNameView.setVisibility(View.GONE);
            bossHealthView.setVisibility(View.GONE);

            if (progress != null) {
                LayoutInflater inflater = LayoutInflater.from(getContext());
                if (progress.getCollect() != null) {
                    for (QuestProgressCollect collect : progress.getCollect()) {
                        QuestCollect contentCollect = quest.getCollectWithKey(collect.key);
                        if (contentCollect == null) {
                            continue;
                        }
                        View view = inflater.inflate(R.layout.quest_collect, collectionContainer, false);
                        SimpleDraweeView iconView = (SimpleDraweeView) view.findViewById(R.id.icon_view);
                        TextView nameView = (TextView) view.findViewById(R.id.name_view);
                        ValueBar valueView = (ValueBar) view.findViewById(R.id.value_view);
                        DataBindingUtils.INSTANCE.loadImage(iconView, "quest_" + quest.getKey() + "_" + collect.key);
                        nameView.setText(contentCollect.text);
                        valueView.set(collect.count, contentCollect.count);

                        collectionContainer.addView(view);
                    }
                }
            }
        }
    }

}

File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/views/social/OldQuestProgressView.kt
package com.habitrpg.android.habitica.ui.views.social

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.LinearLayout
import com.habitrpg.android.habitica.R
import com.habitrpg.android.habitica.databinding.QuestCollectBinding
import com.habitrpg.android.habitica.databinding.QuestProgressOldBinding
import com.habitrpg.android.habitica.models.inventory.QuestContent
import com.habitrpg.android.habitica.models.inventory.QuestProgress
import com.habitrpg.android.habitica.models.inventory.QuestProgressCollect
import com.habitrpg.common.habitica.extensions.dpToPx
import com.habitrpg.common.habitica.extensions.layoutInflater
import com.habitrpg.common.habitica.extensions.loadImage
import com.habitrpg.common.habitica.extensions.setScaledPadding

class OldQuestProgressView : LinearLayout {
    private val binding = QuestProgressOldBinding.inflate(context.layoutInflater, this)

    constructor(context: Context) : super(context) {
        setupView(context)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        setupView(context)
    }

    private fun setupView(context: Context) {
        orientation = VERTICAL
        binding.bossHealthView.valueSuffix = "HP"
        binding.bossRageView.valueSuffix = context.getString(R.string.rage)
        binding.bossHealthView
        setScaledPadding(context, 16, 16, 16, 16)
    }

    fun setData(
        quest: QuestContent,
        progress: QuestProgress?
    ) {
        binding.collectionContainer.removeAllViews()
        if (quest.isBossQuest) {
            binding.bossNameView.text = quest.boss?.name
            binding.bossHealthView.barHeight = 5.dpToPx(context)
            if (progress != null) {
                binding.bossHealthView.set(progress.hp, quest.boss?.hp?.toDouble() ?: 0.0)
            }
            if (quest.boss?.hasRage == true) {
                binding.bossRageWrapper.visibility = VISIBLE
                binding.bossRageView.barHeight = 5.dpToPx(context)
                binding.bossRageView
                binding.bossRageView.set(progress?.rage ?: 0.0, quest.boss?.rage?.value ?: 0.0)
                binding.bossRageNameView.text = quest.boss?.rage?.title
            } else {
                binding.bossRageWrapper.visibility = GONE
            }
            binding.bossHealthWrapper.visibility = VISIBLE
        } else {
            binding.bossHealthWrapper.visibility = GONE
            binding.bossRageWrapper.visibility = GONE

            if (progress != null) {
                val inflater = LayoutInflater.from(context)
                for (collect in progress.collect ?: emptyList<QuestProgressCollect>()) {
                    val contentCollect = quest.getCollectWithKey(collect.key) ?: continue
                    val collectBinding =
                        QuestCollectBinding.inflate(inflater, binding.collectionContainer, true)
                    collectBinding.iconView.loadImage("quest_" + quest.key + "_" + collect.key)
                    collectBinding.nameView.text = contentCollect.text
                    collectBinding.valueView.set(
                        collect.count.toDouble(),
                        contentCollect.count.toDouble()
                    )
                    collectBinding.valueView.barHeight = 5.dpToPx(context)
                }
            }
        }
    }
}
